app/models/game.rb

```diff
@@ -35,40 +35,59 @@ def populate_board!
     Queen.create(player_id: self.white_player_id, game_id: self.id, x_position: 3, y_position: 7)
     King.create(player_id: self.white_player_id, game_id: self.id, x_position: 4, y_position: 7)
   end
-  
+
   has_many :pieces
   has_many :players
   belongs_to :white_player, class_name: 'Player'
   belongs_to :black_player, class_name: 'Player'
-  
+
   def is_piece_present?(x, y)
     self.pieces.where(x_position: x, y_position: y).any?
   end
-  
+
   def get_piece(x, y)
     self.pieces.where(x_position: x, y_position: y).take
   end
-  
+
   def create_timers(time_left)
     self.timers.create(player_id: self.white_player_id,time_left: time_left)
     self.timers.create(player_id: self.black_player_id,time_left: time_left)
   end
-  
+
   def update_timer(current_player)
     timer = self.timers.where(player_id: nil).last
     if timer != nil
       timer.player_id = current_player.id
       timer.save
     end
   end
-  
+
   def check?(player)
-    king = pieces.where(type: 'King', player: player).first
+    king = pieces.where(type: 'King', player: player).last
     opponents_pieces = pieces.where(player: opponent_player(player))
-    
+
     opponents_pieces.any? { |piece| piece.valid_move?(king.x_position, king.y_position) }
   end
-  
+
+  def checkmate?(player)
+    checkmate = true
+    king = pieces.where(type: 'King', player: player).last
```

Comment:
The player also needs to be currently in check, right? Maybe add a conditional to only go through all this code if the player is already in check.